pkg/replica/hash.go

```diff
@@ -120,23 +123,34 @@ func (t *SnapshotHashJob) Execute() (err error) {
 				return
 			}
 
-			SetSnapshotHashInfoToChecksumFile(t.SnapshotName, &xattrType.SnapshotHashInfo{
+			if err = SetSnapshotHashInfoToChecksumFile(t.SnapshotName, &xattrType.SnapshotHashInfo{
 				Method:            defaultHashMethod,
 				Checksum:          checksum,
 				ChangeTime:        changeTime,
 				LastHashedAt:      lastHashedAt,
 				SilentlyCorrupted: silentlyCorrupted,
-			})
+			}); err != nil {
+				logrus.WithError(err).Errorf("failed to set snapshot %s hash info to checksum file", t.SnapshotName)
+				t.State = ProgressStateError
+				t.Error = err.Error()
+				return
+			}
 
-			remain, err := t.isChangeTimeRemain(changeTime)
+			var remain bool
+			remain, err = t.isChangeTimeRemain(changeTime)
 			if !remain {
 				if err == nil {
 					err = fmt.Errorf("snapshot %v modification time is changed", t.SnapshotName)
 				}
 				// Do the best to delete the useless checksum file.
 				// The deletion failure is acceptable, because the mismatching timestamps
 				// will trigger the rehash in the next hash request.
-				DeleteSnapshotHashInfoChecksumFile(t.SnapshotName)
+				if deleteErr := DeleteSnapshotHashInfoChecksumFile(t.SnapshotName); deleteErr != nil {
+					logrus.WithError(deleteErr).Errorf("failed to delete snapshot %v hash info checksum file", t.SnapshotName)
+					t.State = ProgressStateError
+					t.Error = deleteErr.Error()
+					return
+				}
```

Comment:
I'm thinking if we need to error out here.
The operations are best-effort.
Can we switch them to warning level messages?

Reply:
Do you mean we can also remove `t.Error = err.Error()` for these two errors and only need to log them?

Reply:
Yes, just keep `logrus.WithError(deleteErr).Errorf("failed to delete snapshot %v hash info checksum file", t.SnapshotName)`

Reply:
Updated it. Thanks.